sklearn/linear_model/coordinate_descent.py

```diff
@@ -1016,7 +1015,7 @@ def fit(self, X, y):
             # Let us not impose fortran ordering or float64 so far: it is
             # not useful for the cross-validation loop and will be done
             # by the model fitting itself
-            X = check_array(X, 'csc', copy=False)
+            X = check_array(X, 'csc', copy=False, dtype=np.float64)
```

Comment:
True. I wonder if we should add another option to ``check_array`` or if that will get to complicated.